config/mismatches.php

```diff
@@ -17,6 +17,10 @@
         'wikidata_value' => [
             'max_length' => 1500 // Longest allowed value on wikidata
         ],
+        'meta_wikidata_value' => [
+            'max_length' => 100,
+            'format' => '/^Q\d+$/i'
```

Comment:
Q0 shouldn’t be allowed; also, I don’t see why we would want to allow e.g. `q123`, so I’d remove the `i` flag as well.

Also, in `ItemId::PATTERN`, we terminate the pattern with `\z` instead of `$` , which [apparently](https://www.php.net/manual/en/regexp.reference.escape.php) means that it will only match at the end of the full input, even in multi-line mode. I’m not sure where this regex is evaluated, and with which flags, so I think it might be a good idea to use `\z` to make sure it only matches at the end of input. (I guess for the same reason we should also use `\A` instead of `^`? I don’t know why `ItemId::PATTERN` doesn’t do that, to be honest.)

Reply:
If we change the `ItemId::PATTERN`, we will change the current validation and we have the same `/i` flag within `guid` field too. When `guid` is parsed, we use `strtoupper` to make store `item_id` correctly, but we don't do it on `guid` itself. We have different uses of these regular expressions. I'm changing this one as you suggest but I think we should talk about others as well.

Reply:
Yeah, we should only change this one regex. Statement IDs can contain lowercase item IDs, so those should still be allowed.